app/components/layout/sidebar/parent-nav-item.tsx

```diff
@@ -0,0 +1,115 @@
+import { NavLink, useNavigate } from "@remix-run/react";
+import { ChevronDownIcon } from "lucide-react";
+import invariant from "tiny-invariant";
+import {
+  Collapsible,
+  CollapsibleContent,
+  CollapsibleTrigger,
+} from "~/components/shared/collapsible";
+import When from "~/components/when/when";
+import {
+  useIsAnyRouteActive,
+  useIsRouteActive,
+} from "~/hooks/use-is-route-active";
+import type {
+  ChildNavItem,
+  ParentNavItem,
+} from "~/hooks/use-sidebar-nav-items";
+import { tw } from "~/utils/tw";
+import {
+  SidebarMenuButton,
+  SidebarMenuItem,
+  SidebarMenuSub,
+  SidebarMenuSubButton,
+  SidebarMenuSubItem,
+} from "./sidebar";
+
+type ParentNavItemProps = {
+  route: ParentNavItem;
+  tooltip: React.ComponentProps<typeof SidebarMenuButton>["tooltip"];
+  closeIfMobile?: () => void;
+};
+
+export default function ParentNavItem({
+  route,
+  tooltip,
+  closeIfMobile,
+}: ParentNavItemProps) {
+  const navigate = useNavigate();
+  const isAnyChildActive = useIsAnyRouteActive(
+    route.children.map((child) => child.to)
+  );
+
+  const firstChildRoute = route.children[0];
+  invariant(
```

Comment:
How will that actually behave. I never use it so I am not sure what will happen when called like this in the component.

Reply:
This will act as a guard here so if you create a parent nav item but forgot to add any child item in it then it will throw an error that you should add at least one child.